Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -3123,13 +3123,23 @@ static bool isContextMenuEvent(const WebMouseEvent& event)
         return false;
     }
 
-    NSString *plainString = [m_currentSelection string];
-    NSString *htmlString = [m_currentSelection html];
-    NSData *webArchiveData = [m_currentSelection webArchive];
+    NSMutableArray *types = [NSMutableArray array];
+    NSMutableArray *items = [NSMutableArray array];
 
-    NSArray *types = @[ PasteboardTypes::WebArchivePboardType, NSPasteboardTypeString, NSPasteboardTypeHTML ];
+    if (NSData *webArchiveData = [m_currentSelection webArchive]) {
+        [types addObject:PasteboardTypes::WebArchivePboardType];
+        [items addObject:webArchiveData];
+    }
+
+    if (NSData *plainStringData = [[m_currentSelection string] dataUsingEncoding:NSUTF8StringEncoding]) {
+        [types addObject:NSPasteboardTypeString];
+        [items addObject:plainStringData];
+    }
 
-    NSArray *items = @[ webArchiveData, [plainString dataUsingEncoding:NSUTF8StringEncoding], [htmlString dataUsingEncoding:NSUTF8StringEncoding] ];
+    if (NSData *htmlStringData = [[m_currentSelection html] dataUsingEncoding:NSUTF8StringEncoding]) {
```

Comment:
Do we need to also consider if `[m_currentSelection html]` returns `nullptr`?

Reply:
That's fine, because we then message a `nil` object, so the final result will be `nil` too.